### DISCUSSION TITLE
Custom environment variables for worker thread.

### DISCUSSION THREAD

**Discussion by enricoschaaf** [denoland/deno#20064] [Q&A]
Hi 👋 I would like to spawn a worker with custom env variables, is this possible? And if not, is there a way to do this with rust_runtime?

**Comment by boywithkeyboard** [comment]
@enricoschaaf, I don't know of such a feature. This might also be due to the fact that Deno's workers threads follows the [Web Workers API](https://developer.mozilla.org/en-US/docs/Web/API/Worker/Worker) and they don't support environment variables as well as they target browsers.

I'd recommend you to join the Discord server and ask your question there as the people there respond much quicker and might have more in-depth knowledge of the inner and outer parts of Deno. Here's a invite to the Discord server: https://discord.gg/deno

Edit: There's also a open discussion (https://github.com/denoland/deno/discussions/18930) about having a different cwd for each worker thread. As environment variables are per-process they aren't editable on a per-worker basis.
